dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -871,7 +887,30 @@ public Set<Entry> getStarredWorkflows(@ApiParam(hidden = true) @Parameter(hidden
     public Set<Entry> getStarredServices(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User user) {
         User u = userDAO.findById(user.getId());
         checkNotNullUser(u);
-        return u.getStarredEntries().stream().filter(element -> element instanceof Service)
+        return this.getStarredEntries(u, EntryType.SERVICE);
+    }
+
+    /**
+     * Helper function for getStarred endpoints
+     *
+     * @param user
+     * @param entryType
+     * @return a user's starredEntries for desired entryType
+     */
+    public Set<Entry> getStarredEntries(User user, EntryType entryType) {
+        Predicate<Entry> checkInstance;
+
+        if (entryType == EntryType.WORKFLOW) {
+            checkInstance = BioWorkflow.class::isInstance;
+        } else if (entryType == EntryType.SERVICE) {
+            checkInstance = Service.class::isInstance;
+        } else if (entryType == EntryType.NOTEBOOK) {
+            checkInstance = Notebook.class::isInstance;
+        } else { // else would either be Tool or AppTool
+            checkInstance = element -> (element instanceof Tool || element instanceof AppTool);
+        }
+
+        return user.getStarredEntries().stream().filter(checkInstance::test)
```

Comment:
In general, it's good to avoid chains of `if`s like above.  Let the types and objects do the work.  

In that vein, each `Entry` contains its type, so we might pass this method a set of desired `EntryType`s, and then filter like so (this code might not compile :) ):
```
return user.getStarredEntries().stream().filter(entry -> desiredTypes.contains(entry.getEntryType())).collect(...)
```

Then, you could:

```
getStarredEntries(user, Set.of(EntryType.WORKFLOW))
getStarredEntries(user, Set.of(EntryType.TOOL, EntryType.APPTOOL))
// etc
```

Reply:
Thanks Steve for the suggestion! It improved my code alot :)